src/Model/DataModel/StreamData/HoldEmailData.php

```diff
@@ -102,13 +103,13 @@ public function fixPatronName(Patron $patron): string
             $name = ucfirst(strtolower($fullName[1])) . " " . ucfirst(strtolower($fullName[0]));
             return $name;
         } else {
-            throw new APIException(
-                'No names',
-                'Patron did not provide any names',
-                0,
+            throw new NonRetryableException(
+                'Not Acceptable: No names for patron',
+                'Not Acceptable: No names for patron',
```

Comment:
Not sure you need this as debugInfo for the exception since it doesn't add value. You can probably set as null unless you want to add something different here. Check the APIException constructor.